heuristic/delta_calculations.py

```diff
@@ -417,4 +417,127 @@ def regret_objective_function(state, employee, off_shifts, saturdays, days, L_C_
             - 10 * state.hard_vars["delta_positive_contracted_hours"].get(employee, 0)
 
             - 100 * competency_score
+    )
+
+"""
+        What is important to calculate?
+            1. deviation_from_demand vil ikke ha noe å si. Derfor unyttig å kalkulere
+           
+            8. Shift to demand burde ikke trenges ettersom vi sjekker om en ansatt jobber den dagen. 
+            9. Cover multiple demand periods er likt den over. Burde ikke trenge å sjekkes. 
+            10. More than one shift per day burde ikke trenge det heller ettersom vi sjekker et annet sted.
+            11. above/below ikke nødvendig det heller. 
+            
+            13. Dette er opp mot objektivverdi og trenger da sannsynligvis ikke å regne egen objektivverdi. Kan bare se på hver ansatt. 
+        """
+"""
+            2. weekly rest. Når vi setter inn et skift så bør denne kalkuleres. Bør derimot strengt tatt ikke sette verdien, men heller regne ut høyeste mulige verdi.
+            3. Partial weekend vil ha noe å si hvis shiftet ble satt på en lørdag eller søndag
+            4. isolated working day er vanskelig å enkelt kalkulere uten å se på dagen før og dagen etter
+            5. Samme som over. Må sjekke dagen før og dagen etter.
+            6. Consecutive days må sjekke strengt tatt L_C_D - 1 dager før og L_C_D -1 dager etter. 
+            7. Negative deviation from contracted hours. Dette går fort å regne ut. Bare ta nåværende - varighet på skift. Alt som blir minus vil da også bryte hard restriksjon.
+            12. daily_rest bør sjekkes. Bare sjekke den dagen. 
+"""
+
+def employee_shift_value(state, e, shift, saturdays, sundays, invalid_shifts, shift_combinations_violating_daily_rest, shift_sequences_violating_daily_rest, shifts_at_week, weeks, shifts_at_day, week, competency_score):
+    day = int(shift[0]/24)
+    daily_rest_error = regret_daily_rest_error(state, day, e, shift, invalid_shifts, shift_combinations_violating_daily_rest, shift_sequences_violating_daily_rest)
+    weekly_rest_error = regret_weekly_rest(state, shifts_at_week, e, week, shift)
+    partial_weekend_error = regret_partial_weekend(state, e, shifts_at_day, saturdays, sundays, day)
+    isolated_days_error = regret_isolated_days(state, e, shifts_at_day, day, weeks)
+    deviation_contracted_hours = regret_deviation_contracted_hours(state, e, shift, week, weeks)
+
+    return (
+            daily_rest_error 
+            + weekly_rest_error 
+            + partial_weekend_error 
+            + isolated_days_error 
+            + deviation_contracted_hours 
+            - 100 * competency_score
             )
+
+def regret_weekly_rest(state, shifts_at_week, e, week, shift):
+    actual_shifts = [(t, v) for t,v in shifts_at_week[week] if state.x[e,t,v] == 1 or (t,v) == (shift[0],shift[1])]
+    off_shift_periods = []
+    important = [7 * 24 * week, 7 * 24 * (week + 1) ]
```

Comment:
What is the purpose of this list? It contains the start and end time for the week?

Reply:
Yes

Reply:
I mostly reused code from the other calculate weekly rest. It is possible to use the value directly since we are only calculating one week at a time. If we would expand this function to work with multiple weeks, then this list is necessary

Reply:
Could you rename it to _week_interval_  or something more descriptive? I had a hard time understanding what this variable meant farther down.